atef/report.py

```diff
@@ -0,0 +1,546 @@
+"""
+Report rendering framework
+"""
+
+import hashlib
+from dataclasses import fields
+from datetime import datetime
+from pathlib import Path
+from typing import Any, Dict, Generator, List, Optional, Tuple, Union
+
+from reportlab import platypus
+from reportlab.lib import colors, enums, pagesizes, units
+from reportlab.lib.styles import ParagraphStyle as PS
+from reportlab.lib.styles import getSampleStyleSheet
+from reportlab.lib.units import cm
+from reportlab.pdfgen.canvas import Canvas
+from reportlab.platypus.doctemplate import BaseDocTemplate, PageTemplate
+from reportlab.platypus.flowables import Flowable
+from reportlab.platypus.frames import Frame
+from reportlab.platypus.paragraph import Paragraph
+from reportlab.platypus.tableofcontents import TableOfContents
+
+from atef.check import Result
+from atef.config import (PreparedComparison, PreparedConfiguration,
+                         PreparedDeviceConfiguration, PreparedFile,
+                         PreparedGroup, PreparedPVConfiguration,
+                         PreparedSignalComparison, PreparedToolComparison,
+                         PreparedToolConfiguration)
+from atef.enums import Severity
+
+h1 = PS(name='Heading1', fontSize=16, leading=20)
```

Comment:
In the future, all these style things could all be part of another configuration file setting, I'd imagine